contracts/integrations/custom/CustomIntegrationIdleFinance.sol

```diff
@@ -0,0 +1,338 @@
+     * @return address[] memory           List of reward token addresses
+     */
+    function _getRewardTokens(
+        address /* _data */
+    ) internal pure override returns (address[] memory) {
+        // No extra rewards. All get consolidated to the vault input token on exit
+        return new address[](1);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * The tokens to be purchased by the strategy on enter according to the weights.
+     * Weights must add up to 1e18 (100%)
+     *
+     * hparam  _data                      Address provided as param
+     * @return _inputTokens               List of input tokens to buy
+     * @return _inputWeights              List of weights for the tokens to buy
+     */
+    function getInputTokensAndWeights(
+        bytes calldata _data
+    ) external view override returns (address[] memory _inputTokens, uint256[] memory _inputWeights) {
+        // Vault takes 100% in the vault token
+        address vault = BytesLib.decodeOpDataAddress(_data);
+        address[] memory inputTokens = new address[](1);
+        inputTokens[0] = IIdleToken(vault).token();
+        uint256[] memory inputWeights = new uint256[](1);
+        inputWeights[0] = 1e18; // 100%
+        return (inputTokens, inputWeights);
+    }
+
+    /**
+     * The tokens to be received on exit.
+     *
+     * hparam  _strategy                  Strategy address
+     * hparam  _data                      Bytes data
+     * hparam  _liquidity                 Number with the amount of result tokens to exit
+     * @return exitTokens                 List of output tokens to receive on exit
+     * @return _minAmountsOut             List of min amounts for the output tokens to receive
+     */
+    function getOutputTokensAndMinAmountOut(
+        address _strategy,
+        bytes calldata _data,
+        uint256 /* _liquidity */
+    ) external view override returns (address[] memory exitTokens, uint256[] memory _minAmountsOut) {
+        // Vault exits 100% in the vault token
+        address vault = BytesLib.decodeOpDataAddress(_data);
+        address[] memory outputTokens = new address[](1);
+        outputTokens[0] = IIdleToken(vault).token();
+        uint256[] memory outputAmounts = new uint256[](1);
+        outputAmounts[0] = IIdleToken(vault).tokenPriceWithFee(_strategy);
+        return (outputTokens, outputAmounts);
+    }
+
+    /**
+     * The price of the result token based on the asset received on enter
+     *
+     * hparam  _data                      Bytes data
+     * hparam  _tokenDenominator          Token we receive the capital in
+     * @return uint256                    Amount of result tokens to receive
+     */
+    function getPriceResultToken(
+        bytes calldata _data,
+        address _tokenDenominator
+    ) external view override returns (uint256) {
+        address vault = BytesLib.decodeOpDataAddress(_data);
```

Comment:
Copied from Yearn example, does this need to be here though?

Reply:
Yes, and you need to multiply this price by the Idle Token

Idle Token -> Amount of Underlying -> Amount of reserve asset

```
      return IIdleToken(vault).tokenPrice().preciseMul(_getPrice(IIdleToken(vault).token(), _tokenDenominator));
```